TicTacToeApp/Board.swift

```diff
@@ -0,0 +1,48 @@
+
+public class Board {
+  
+  private let markedSpaces: [Int: PlayerMark]
+  private let size = 3
+
+  public init() {
+    markedSpaces = [:]
+  }
+
+  public init(markedSpaces: [Int : PlayerMark]) {
+    self.markedSpaces = markedSpaces
+  }
+
+  public func markSpace(id: Int, mark: PlayerMark) -> Board {
+    var newSpaces = markedSpaces
+    newSpaces[id] = mark
+
+    return Board(markedSpaces: newSpaces)
+  }
+
+  public func readSpace(id: Int) -> PlayerMark? {
+    return markedSpaces[id]
+  }
+
+  public func allSpacesMarked() -> Bool {
+    return (0..<(size * size)).reduce(true, combine: { $0 && markedSpaces[$1] != nil })
+  }
+
+  public func hasWinningLine() -> Bool {
```

Comment:
Given that it implies knowledge of game rules? I'm with you there, but it also strikes me as an issue of semantics - for example, I could rename it to allSameMarkInALine() or hasThreeInARow() or something like that, which seems like a reasonable board query to me. Would renaming the method resolve the issue?